pyomo/contrib/pynumero/cmake/sparse_utils/src/SparseUtils.cpp

```diff
@@ -257,6 +258,58 @@ void sym_csc_diagonal(int* col_pointer,
     }
 }
 
+int sym_csr_allnnz(int* row_pointer,
+                   int* col_indices,
+                   int nrows)
+{
+    int row, col;
+    int off_diagonals = 0;
+    int on_diagonals = 0;
+
+    for (int i=0; i<nrows; ++i)
+    {
+        row = i;
+        for(int jj = row_pointer[row]; jj < row_pointer[row+1]; ++jj)
+        {
+            col = col_indices[jj];
+            if(col==row){
+                ++on_diagonals;
+            }
+            else{
+                ++off_diagonals;
+            }
+        }
+    }
+    return 2*off_diagonals + on_diagonals;
+}
+
+int sym_csc_allnnz(int* col_pointer,
+                   int* row_indices,
+                   int ncols)
```

Comment:
Technically, this method and `sym_csr_allnnz` are identical.  Why not remove this one and rename `sym_csr_allnnz` to `sym_allnnz`?

Reply:
Removed SparseUtils from `pynumero.cmake` because is no longer used.